libs/src/services/creatorNode/CreatorNodeSelection.js

```diff
@@ -0,0 +1,249 @@
+const semver = require('semver')
+
+const ServiceSelection = require('../../service-selection/ServiceSelection')
+const { timeRequest } = require('../../utils/network')
+const { CREATOR_NODE_SERVICE_NAME } = require('./constants')
+
+class CreatorNodeSelection extends ServiceSelection {
+  constructor ({ creatorNode, numberOfNodes, ethContracts, whitelist, blacklist }) {
+    super({
+      getServices: async () => {
+        this.currentVersion = await ethContracts.getCurrentVersion(CREATOR_NODE_SERVICE_NAME)
+        const services = await this.ethContracts.getServiceProviderList(CREATOR_NODE_SERVICE_NAME)
+        return services.map(e => e.endpoint)
```

Comment:
every one of these endpoints should be valid at the moment this function returns - so sanitized inputs, valid FQDN etc etc. We cannot rely on the `string` value stored on the blockchain when selecting replicas.

Reply:
let's either confirm that is validated in the existing code or add functionality here